# BiotrAIn Course - scikit-bio

This repository hosts **interactive tutorials for using [scikit-bio](http://scikit-bio.org)** in biological and microbiome data analysis. The materials were originally developed for the [ISMB 2024](https://www.iscb.org/ismb2024/programme-schedule/tutorials#ip3) conference tutorial **‚ÄúHarnessing scikit-bio for biological data exploration‚Äù**, and have been adapted and extended for other educational settings.

## üìò About this version

This tutorial series is part of the **BiotrAIn Course: *Concepts, Approaches and Applications of Artificial Intelligence in Bioscience***, organized by [CABANA Network](https://www.cabanabr.org/), and supported by **EMBL-EBI**, the **Chan Zuckerberg Initiative**, and the **University of Costa Rica (UCR)**.

Additionally, this workshop is offered as part of the *‚ÄúVieRnes de Bioinform√°tica en el LIIGH‚Äù* initiative, which provides free bioinformatics training sessions to the Latin American research community.

üóìÔ∏è **Class dates**:  
- Friday, May 23, 2025  
- Friday, May 30, 2025

üìç **Location**:  
University of Costa Rica (UCR), San Jos√©, Costa Rica

---

## üß™ Tutorial Contents

These notebooks are based on and expand upon the official ISMB 2024 scikit-bio tutorial. 

* 01_DataIntegrationProcessing.ipynb
  * Data integration and preprocessing strategies for microbiome data
* 02_UniFrac_BetaDiversity.ipynb 
  * Phylogenetic beta diversity with UniFrac
* 03_BrayCurtisJaccard_BetaDiversity.ipynb
  * Non-phylogenetic beta diversity: Bray-Curtis and Jaccard indices
* 04_ClusteringKmeansDBSCAN.ipynb 
  * Clustering microbial communities: K-means and DBSCAN
* 05_ClusteringEvaluation.ipynb
  * Evaluating clustering performance
* 06_OtherDatasets.ipynb
  * Applying analysis workflows to other biological datasets

---

## üìö References

1. [Protein remote homology detection and structural alignment using deep learning](https://www.nature.com/articles/s41587-023-01917-2)  
2. [ProtTrans: Towards Cracking the Language of Life‚Äôs Code Through Self-Supervised Learning](https://www.biorxiv.org/content/10.1101/2020.07.12.199554v3)  
3. [UniFrac: a New Phylogenetic Method for Comparing Microbial Communities](https://journals.asm.org/doi/10.1128/AEM.71.12.8228-8235.2005)  
4. [Establishing microbial composition measurement standards with reference frames](https://www.nature.com/articles/s41467-019-10656-5)  
5. [Compositional data analysis for high-throughput sequencing datasets](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4030730/)  
6. [Balances: a New Perspective for Microbiome Analysis](https://pubmed.ncbi.nlm.nih.gov/30035234/)  
7. [Compositionally Aware Phylogenetic Beta-Diversity Measures](https://journals.asm.org/doi/10.1128/msystems.00050-22)  
